tests/test_glue.py

```diff
@@ -216,6 +214,57 @@ def test_glue_job_untag(self):
         self.assertEqual(tags.get('Tags'), {})
         self.assertEqual(len(resources), 1)
 
+    def test_glue_crawler_tag(self):
+        session_factory = self.replay_flight_data("test_crawler_tags")
+        client = session_factory().client("glue")
+
+        policy = {
+            'name': 'test',
+            'resource': 'glue-crawler',
+            'filters': [{'tag:abcd': 'absent'}],
+            'actions': [
+                {
+                    'type': 'tag',
+                    'key': 'abcd',
+                    'value': 'xyz'
+                },
+            ]
+        }
+        p = self.load_policy(
+            policy,
+            config={'account_id': '644160558196'},
+            session_factory=session_factory)
+
+        resources = p.run()
+        arn = p.resource_manager.generate_arn(resources[0]['Name'])
```

Comment:
use get_arns, generate_arn is a lower level interface not a public one per se. get_arns will delegate to generate_arn, so this will test the outcome, but the internals of generate_arn are an implementation detail, and one thats going to change soon.

Reply:
i'll handle this in the arn refactor branch